# 01 — Smoke test del entorno
**Objetivo:** validar que el entorno (venv `.venv-yelpdm`, Java, Spark) funciona y que la data bronze está accesible, antes de la ingesta real.

Checklist:
1. Versiones correctas (Python 3.10–3.12, PySpark 3.5.x)
2. SparkSession levanta en local
3. Lectura de `business.json` y conteo de registros (esperado: **150,346**)

In [1]:
import sys
sys.path.append("..")  # para importar src/ desde notebooks/

import pyspark, numpy, pandas, duckdb
print(f"Python  {sys.version.split()[0]}")
print(f"PySpark {pyspark.__version__}")
print(f"numpy {numpy.__version__} | pandas {pandas.__version__} | duckdb {duckdb.__version__}")

Python  3.11.6
PySpark 3.5.8
numpy 2.4.6 | pandas 3.0.3 | duckdb 1.5.3


In [2]:
from src.config import YELP, ensure_dirs, spark_session

ensure_dirs()
spark = spark_session("smoke-test")
print("Spark OK ->", spark.version)

26/06/18 13:50:22 WARN Utils: Your hostname, MacBook-Pro-de-johar.local resolves to a loopback address: 127.0.0.1; using 192.168.100.41 instead (on interface en0)
26/06/18 13:50:22 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).


26/06/18 13:50:22 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable


Spark OK -> 3.5.8


In [3]:
# Lectura del archivo más pequeño (114 MB) como prueba de punta a punta
biz = spark.read.json(str(YELP["business"]))
print("Negocios:", biz.count())  # esperado: 150346
biz.printSchema()

Negocios: 150346
root
 |-- address: string (nullable = true)
 |-- attributes: struct (nullable = true)
 |    |-- AcceptsInsurance: string (nullable = true)
 |    |-- AgesAllowed: string (nullable = true)
 |    |-- Alcohol: string (nullable = true)
 |    |-- Ambience: string (nullable = true)
 |    |-- BYOB: string (nullable = true)
 |    |-- BYOBCorkage: string (nullable = true)
 |    |-- BestNights: string (nullable = true)
 |    |-- BikeParking: string (nullable = true)
 |    |-- BusinessAcceptsBitcoin: string (nullable = true)
 |    |-- BusinessAcceptsCreditCards: string (nullable = true)
 |    |-- BusinessParking: string (nullable = true)
 |    |-- ByAppointmentOnly: string (nullable = true)
 |    |-- Caters: string (nullable = true)
 |    |-- CoatCheck: string (nullable = true)
 |    |-- Corkage: string (nullable = true)
 |    |-- DietaryRestrictions: string (nullable = true)
 |    |-- DogsAllowed: string (nullable = true)
 |    |-- DriveThru: string (nullable = true)
 |    |-- Go

In [4]:
spark.stop()
print("Entorno validado ✅")

Entorno validado ✅


## Resumen del notebook

Validamos el entorno de trabajo: versiones correctas (Python 3.11, PySpark 3.5, Java 17), SparkSession operativa en local y lectura completa de `business.json` con los **150,346 negocios** esperados. Con la infraestructura verificada, continuamos en `02_ingesta_bronze_a_silver.ipynb` con la construcción de la capa silver.